server/app/scanpy_engine/scanpy_engine.py

```diff
@@ -22,17 +22,48 @@
 
 class ScanpyEngine(CXGDriver):
 
-    def __init__(self, data, layout_method=None, diffexp_method=None, max_category_items=100):
+    def __init__(self, data, layout_method=None, diffexp_method=None,
+                 obs_names=None, var_names=None, max_category_items=100):
         super().__init__(data, layout_method=layout_method, diffexp_method=diffexp_method,
                          max_category_items=max_category_items)
-        self._validatate_data_types()
-        self._add_mandatory_annotations()
+        self._alias_annotation_names(obs_names, var_names)
+        self._validate_data_types()
         self.cell_count = self.data.shape[0]
         self.gene_count = self.data.shape[1]
         self.layout_options = ["umap", "tsne"]
         self.diffexp_options = ["ttest"]
         self._create_schema()
 
+    def _alias_annotation_names(self, obs_names, var_names):
```

Comment:
As discussed, change the signature to make what the name params are clearer.

Reply:
done